barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -464,6 +464,122 @@ template <typename Curve_> class IPA {
         // Check if C_right == C₀
         return (C_zero.normalize() == right_hand_side.normalize());
     }
+    /**
+     * @brief  Recursively verify the correctness of an IPA proof. Unlike native verification, there is no
+     * parallelisation in this function as our circuit construction does not currently support parallelisation.
+     *
+     * @details  batch_mul is used instead of pippenger as pippenger is not implemented to be used in stdlib context for
+     * now and under the hood we perform bigfield to cycle_scalar conversions for the batch_mul. That is because
+     * cycle_scalar has very reduced functionality at the moment and doesn't support basic arithmetic operations between
+     * two cycle_scalar operands (just for one cycle_group and one cycle_scalar to enable batch_mul).
+     * @param vk
+     * @param opening_claim
+     * @param transcript
+     * @return VerifierAccumulator
+     * @todo (https://github.com/AztecProtocol/barretenberg/issues/1018): simulator should use the native verify
+     * function with parallelisation
+     */
+    static VerifierAccumulator reduce_verify_internal(const std::shared_ptr<VK>& vk,
+                                                      const OpeningClaim<Curve>& opening_claim,
+                                                      auto& transcript)
+        requires Curve::is_stdlib_type
+    {
+        // Step 1.
+        // Receive polynomial_degree + 1 = d from the prover
+        const auto poly_length = static_cast<uint32_t>(
+            transcript->template receive_from_prover<typename Curve::BaseField>("IPA:poly_degree_plus_1")
```

Comment:
Add a todo to actually assert_equal the value received from the prover. Right now it is a spare value to bruteforce challenges

Reply:
you mean with the circuit size given in the verification_key or what?